Looks great! I have tested it out, and I just have one single request. Please remove the console.log on line 292 of Injected.js. Not only is it not doing anything useful, but it is being flagged as an error by the linter:

    C:\Users\DHentschel\Dropbox\CodinGame\Chrome Extensions\CGSpunk\app\scripts\Injected.js
      292:21  error  Strings must use singlequote  quotes

    ✖ 1 problem (1 error, 0 warnings)

Otherwise, I have done some minimal testing, and everything that I tried works just fine. Make this change and I will accept the pull request and publish the results.

Also, what is your CG username so that I can give you credit? :-)

 - danBhentschel

--- a/app/scripts/Injected.js
+++ b/app/scripts/Injected.js
@@ -289,8 +289,6 @@ var __CGSpunk_Injected =
 
     function getMyAgentFromGameManager(gameManager, me) {
         let myAgents = gameManager.currentGameInfo.agents.filter(_ => _.typeData.me === true);
-        console.log(`myAgents`);
-
         if (myAgents.length === 0) return null;
         if (myAgents.length === 1) return myAgents[0];
         let ideAgents = myAgents.filter(_ => _.agentId === -1);

--- a/app/scripts/Injected.js
+++ b/app/scripts/Injected.js
@@ -345,7 +345,11 @@ var __CGSpunk_Injected =
             !(angular.element('.player').scope().gameManager)) {
 	*/
 	// SixK : We want to be sure typeData is not undefined, probably calculated after receiving all other data
-        if (!(angular.element('cg-player-sandbox').parent().scope().api.gameManagerAdapter) ||
+        if (!(angular.element('cg-player-sandbox')) ||
+            !(angular.element('cg-player-sandbox').parent()) ||
+            !(angular.element('cg-player-sandbox').parent().scope()) ||
+            !(angular.element('cg-player-sandbox').parent().scope().api) ||
+	    !(angular.element('cg-player-sandbox').parent().scope().api.gameManagerAdapter) ||
             !(angular.element('cg-player-sandbox').parent().scope().api.gameManagerAdapter.currentGameInfo) ||
             !(angular.element('cg-player-sandbox').parent().scope().api.gameManagerAdapter.currentGameInfo.agents[0]) ||
             !(angular.element('cg-player-sandbox').parent().scope().api.gameManagerAdapter.currentGameInfo.agents[0].typeData) ||  

